libraries/pushstore/impl/src/main/kotlin/io/element/android/libraries/pushstore/impl/UserPushStoreDataStore.kt

```diff
@@ -37,7 +40,24 @@ class UserPushStoreDataStore(
     private val context: Context,
     userId: SessionId,
 ) : UserPushStore {
-    private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = "push_store_$userId")
+    // Hash the sessionId to get ride of exotic chars and take only the first 16 chars.
+    // The risk of collision is not high.
+    private val preferenceName = "push_store_${userId.value.hash().take(16)}"
+
+    init {
+        // Migrate legacy data. Previous file can be too long if the userId is too long. The userId can be up to 255 chars.
+        // Example of long file path, with `averylonguserid` replacing a very longer name
+        // /data/user/0/io.element.android.x.debug/files/datastore/push_store_@averylonguserid:example.org.preferences_pb
+        val legacyFile = context.preferencesDataStoreFile("push_store_$userId")
+        if (legacyFile.exists()) {
+            Timber.d("Migrating legacy push data store for $userId")
+            if (!legacyFile.renameTo(context.preferencesDataStoreFile(preferenceName))) {
+                Timber.w("Failed to migrate legacy push data store for $userId")
+            }
+        }
```

Comment:
Done in the main thread, but maybe not a big deal?

Reply:
Yes, I think it's acceptable. I wanted to include this in the `by preferencesDataStore(name = preferenceName)`, but did not find a good way to do it.